experimental/uefi/app/src/serial.rs

```diff
@@ -71,24 +71,16 @@ impl<'boot> Serial<'boot> {
     }
 }
 
-impl<'boot> ciborium_io::Write for Serial<'boot> {
+impl<'boot> runtime::echo::Interface for Serial<'boot> {
     type Error = Error<()>;
 
-    fn write_all(&mut self, data: &[u8]) -> Result<(), Self::Error> {
+    fn send(&mut self, data: &[u8]) -> Result<(), Self::Error> {
         self.serial.write(data).discard_errdata()
     }
 
-    fn flush(&mut self) -> Result<(), Self::Error> {
-        Ok(())
-    }
-}
-
-impl<'boot> ciborium_io::Read for Serial<'boot> {
-    type Error = Error<()>;
-
     // Try to fill the buffer, ignoring any timeout errors. Any other errors
```

Comment:
pre-existing: why are we ignoring timeout errors?

Reply:
We want to block until we've read a full line, but by default read() will time out every 100 ms or so. Yes, this is inefficient busy-waiting and we'll need to replace it with something better.